homeassistant/components/husqvarna_automower/config_flow.py

```diff
@@ -0,0 +1,35 @@
+"""Config flow to add the integration via the UI."""
+import logging
+from typing import Any
+
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers import config_entry_oauth2_flow
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class HusqvarnaConfigFlowHandler(
+    config_entry_oauth2_flow.AbstractOAuth2FlowHandler,
+    domain=DOMAIN,
+):
+    """Handle a config flow."""
+
+    DOMAIN = DOMAIN
+    VERSION = 1
+
+    async def async_oauth_create_entry(self, data: dict[str, Any]) -> FlowResult:
+        """Create an entry for the flow."""
+        await self.async_set_unique_id(DOMAIN)
```

Comment:
Why do we only allow one config entry?

Reply:
More are possible as well. I've added this function. The unique identifier is now the `auth_implemenation`, a combination of domain and Api-Key. As you can register for multiple Api-Keys in one account (with multiple mowers), there was the need to add the Api-Key to the unique_id of the lawn mower.
The disadavantage is now, if you register with a new Api-Key, there will be new entities now. Is this okay?

Reply:
When you say api key, do you mean client secret? Doesn't the user account get access to all mowers of the user regardless of which application, ie which client id and secret, was set up to access the api?

Reply:
Good. Is there a user id that we can query? If so we could use that as unique_id for the config entry.

Reply:
Yes we can :)
I change it.